sympy/solvers/ode/single.py

```diff
@@ -917,6 +917,93 @@ def _get_general_solution(self, *, simplify_flag: bool = True):
         return [gensol]
 
 
+class RationalRiccati(SinglePatternODESolver):
+    r"""
+    Gives all rational solutions to the first
+    order Riccati Differential Equation
+
+    .. math :: y' = b_0(x) + b_1(x) y + b_2(x) y^2
+
+    where `b_0`, `b_1` and `b_2` are rational functions of `x`
+    with `b_2 \ne 0` (`b_2 = 0` would make it a Bernoulli equation).
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, Function, dsolve, checkodesol
+    >>> f = Function('f')
+    >>> x = Symbol('x')
+
+    >>> eq = f(x).diff(x) - x*f(x)**2
+    >>> sols = dsolve(eq, hint="1st_rational_riccati")
+    >>> sols
+    Eq(f(x), -2/(C1 + x**2))
+    >>> checkodesol(eq, sols)
+    (True, 0)
+
+    >>> eq = x**3*f(x).diff(x) - x**2*f(x) - f(x)**2
+    >>> sols = dsolve(eq, hint="1st_rational_riccati")
+    >>> sols
+    Eq(f(x), C1*x**2/(C1 + x))
+    >>> checkodesol(eq, sols)
+    (True, 0)
+
+    >>> eq = -x**4*f(x)**2 + x**3*f(x).diff(x) + x**2*f(x) + 20
+    >>> sols = dsolve(eq, hint="1st_rational_riccati")
+    >>> sols
+    Eq(f(x), (4*C1 - 5*x**9 - 4)/(x**2*(C1 + x**9 - 1)))
+    >>> checkodesol(eq, sols)
+    (True, 0)
+
+    >>> eq = x**3*f(x).diff(x) - x**2*(f(x) + f(x).diff(x)) + 2*x*f(x) - f(x)**2
+    >>> sols = dsolve(eq, hint="1st_rational_riccati")
+    >>> sols
+    Eq(f(x), x**2*(C1 + 1)/(C1 + x))
+    >>> checkodesol(eq, sols)
+    (True, 0)
+
+    .. [1] Algorithm (Pg 78) - https://www3.risc.jku.at/publications/download/risc_5387/PhDThesisThieu.pdf
+    .. [2] Examples - https://www3.risc.jku.at/publications/download/risc_5197/RISCReport15-19.pdf
+    """
+    has_integral = False
+    hint = "1st_rational_riccati"
+    order = [1]
+
+    def _wilds(self, f, x, order):
+        b0 = Wild('b0', exclude=[f(x), f(x).diff(x)])
+        b1 = Wild('b1', exclude=[f(x), f(x).diff(x)])
+        b2 = Wild('b2', exclude=[f(x), f(x).diff(x)])
+        return (b0, b1, b2)
+
+    def _equation(self, fx, x, order):
+        b0, b1, b2 = self.wilds()
+        return fx.diff(x) - b0 - b1*fx - b2*fx**2
```

Comment:
Other forms that reduce to this like `f'(x) - (f(x) + 1)**2`.

Would those be matched?

Reply:
Yes, since I am always expanding the equation first and collecting the coefficients based on `f(x)`.